examples/simple-chat/src/simple_chat.ts

```diff
@@ -27,51 +27,91 @@ class ChatUI {
   // all requests send to chat are sequentialized
   private chatRequestChain: Promise<void> = Promise.resolve();
 
-  constructor(chat: ChatInterface, localChat: ChatInterface) {
+  private constructor() {
+  }
+
+  /**
+   * An asynchronous factory constructor since we need to await getMaxStorageBufferBindingSize();
+   * this is not allowed in a constructor (which cannot be asynchronous).
+   */
+  public static CreateAsync = async (chat: ChatInterface, localChat: ChatInterface) => {
+    const chatUI = new ChatUI();
     // use web worker to run chat generation in background
-    this.chat = chat;
-    this.localChat = localChat;
+    chatUI.chat = chat;
+    chatUI.localChat = localChat;
     // get the elements
-    this.uiChat = getElementAndCheck("chatui-chat");
-    this.uiChatInput = getElementAndCheck("chatui-input") as HTMLInputElement;
-    this.uiChatInfoLabel = getElementAndCheck("chatui-info-label") as HTMLLabelElement;
+    chatUI.uiChat = getElementAndCheck("chatui-chat");
+    chatUI.uiChatInput = getElementAndCheck("chatui-input") as HTMLInputElement;
+    chatUI.uiChatInfoLabel = getElementAndCheck("chatui-info-label") as HTMLLabelElement;
     // register event handlers
     getElementAndCheck("chatui-reset-btn").onclick = () => {
-      this.onReset();
+      chatUI.onReset();
     };
     getElementAndCheck("chatui-send-btn").onclick = () => {
-      this.onGenerate();
+      chatUI.onGenerate();
     };
     // TODO: find other alternative triggers
     getElementAndCheck("chatui-input").onkeypress = (event) => {
       if (event.keyCode === 13) {
-        this.onGenerate();
+        chatUI.onGenerate();
       }
     };
 
+    // When we detect low maxStorageBufferBindingSize, we assume that the device (e.g. an Android
+    // phone) can only handle small models and make all other models unselectable. Otherwise, the
+    // browser may crash. See https://github.com/mlc-ai/web-llm/issues/209.
+    let restrictModels = false;
+    const lowRequirementModels = new Set<string>([
+      'RedPajama-INCITE-Chat-3B-v1-q4f16_1-1k',
```

Comment:
It seems brittle to have IDs there. Why not having a boolean attribute in the `model_list` for each model instead?

Reply:
i agree we can have a vram_requirement field in model_list

Reply:
Used a boolean to decide in the latest commit. Also added `vram_required` in model_list as well (though not used to decide whether a model should be used for Android, since vram alone may not be enough -- there is also the context length which determines the per-buffer allocation size).